include/swift/AST/AutoDiff.h

```diff
@@ -0,0 +1,92 @@
+//===--- AutoDiff.h - Swift Automatic Differentiation ---------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+//  SWIFT_ENABLE_TENSORFLOW
+//  This file defines AST support for automatic differentiation.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SWIFT_AST_AUTODIFF_H
+#define SWIFT_AST_AUTODIFF_H
+
+#include "ASTContext.h"
+#include "llvm/ADT/SmallBitVector.h"
```

Comment:
SmallBitVector.h looks unused.

Reply:
Also, is ASTContext.h necessary here?  Can you forward declare or include smaller headers?  ASTContext.h includes the world.

Reply:
Thanks for catching these. It turned out, I only needed `IndexSubset.h` instead of `ASTContext.h`